app/manager-frontend/src/pages/ActivitySource/index.jsx

```diff
@@ -113,21 +125,49 @@ export const ActivitySource = ({ activitySource, action, id }) => {
                   name={"ResourceId"}
                   type="ResourceId"
                 />
-                <Button
-                  w="full"
-                  leftIcon={<FaArrowRight />}
-                  colorScheme="blue"
-                  type="submit"
-                  disabled={isSubmitting}
-                  isLoading={isSubmitting}
-                >
-                  {submitText}
-                </Button>
+                <Flex>
+                  <Button
+                    w="full"
+                    leftIcon={<FaArrowRight />}
+                    colorScheme="blue"
+                    type="submit"
+                    disabled={isSubmitting}
+                    isLoading={isSubmitting}
+                    mx={id ? 4 : 0}
+                  >
+                    {submitText}
+                  </Button>
+                  {id && (
+                    <Button w="full" colorScheme="red" onClick={onOpen} mx={4}>
+                      Delete
+                    </Button>
+                  )}
+                </Flex>
               </VStack>
             </Form>
           )}
         </Formik>
       </VStack>
+      <Modal isOpen={isOpen} onClose={onClose}>
+        <ModalOverlay />
+        <ModalContent>
+          <ModalHeader>Delete Activity Source {id}</ModalHeader>
+          <ModalCloseButton />
+          <ModalBody>
+            Are you sure you want to delete this activity source? You will not
+            be able to reverse this
+          </ModalBody>
+
+          <ModalFooter>
+            <Button colorScheme="blue" mr={3} onClick={onClose}>
```

Comment:
🔧 this shouldn't be blue. Generally (especially in a modal or a card, but also generally on pages) you should only have one type of action solidly coloured, to call attention to it.

Cancel / Close / Back etc. should almost never be coloured.
Green is usually creating a new thing
Red is usually something dangerous or destructive (like deleting)
Blue (or the site's main color) is often used for a general most important action.

Reply:
💡 That's an interesting point, @beforan